sw-emulator/lib/periph/src/ml_dsa87.rs

```diff
@@ -341,11 +349,15 @@ impl MlDsa87 {
 
         let signature = &bytes_from_words_le(&self.signature);
 
-        let result = public_key.verify(message, &signature[..SIG_LEN].try_into().unwrap(), &[]);
+        let success = public_key.verify(message, &signature[..SIG_LEN].try_into().unwrap(), &[]);
 
-        self.verify_res
-            .iter_mut()
-            .for_each(|e| *e = if result { 1 } else { 0 });
+        if success {
+            self.verify_res
+                .copy_from_slice(&self.signature[..ML_DSA87_VERIFICATION_SIZE / 4]);
+        } else {
+            self.verify_res
+                .copy_from_slice(&[0; ML_DSA87_VERIFICATION_SIZE / 4]);
```

Comment:
Might be easier to do?
```suggestion
            self.verify_res.fill(0)
```

Reply:
Updated.